### PR TITLE
feat(chainwatch): capture circulating fil per tipset metric

### PR DESCRIPTION
**Pull request by placer14**
Part of https://github.com/filecoin-project/sentinel/issues/38

### CODE REVIEW

File: cmd/lotus-chainwatch/syncer/sync.go [hunk 1]
```diff
@@ -173,6 +182,10 @@ func (s *Syncer) Start(ctx context.Context) {
 						log.Errorw("failed to gather unsynced blocks", "error", err)
 					}
 
+					if err := s.storeCirculatingSupply(ctx, change.Val); err != nil {
+						log.Errorw("failed to store circulating supply", "error", err)
+					}
+
```
Comment (placer14) [inline cmd/lotus-chainwatch/syncer/sync.go:188]: A short term problem of this being attached to the syncer is that only tipsets which arrive while chainwatch is running will be processed here. Follow up work is planned to make sure back tipsets are processed properly.

File: cmd/lotus-chainwatch/syncer/sync.go [hunk 2]
```diff
@@ -264,6 +277,24 @@ func (s *Syncer) syncedBlocks(timestamp time.Time) (map[cid.Cid]struct{}, error)
 	return out, nil
 }
 
+func (s *Syncer) storeCirculatingSupply(ctx context.Context, tipset *types.TipSet) error {
+	supply, err := s.node.StateCirculatingSupply(ctx, tipset.Key())
+	if err != nil {
+		return err
+	}
+
+	ceInsert := `insert into chain_economics (parent_state_root, circulating_fil) values ('%s', '%s');`
```
Comment (frrist) [inline cmd/lotus-chainwatch/syncer/sync.go:286]: any reason to not construct the complete insert statement here?

**Comment by lanzafame** [comment]
@placer14 I am getting the following error when running chainwatch against a calibration net node:
```
2020-08-09T08:09:30.487+1000    ERROR   syncer  syncer/sync.go:186      failed to store circulating supply      {"error": "insert circulating supply for tipset ({bafy2bzaceasvizcp2v3ms6hgs4e6dtuff466ushzmckubgjyxcyhk7ihi5yyk,bafy2bzacedfcucf4yraydee46v2zcexvrpwihl3jb3qcoghbcocezb5s7ttgi}): pq: duplicate key value violates unique constraint \"chain_economics_pk\"", "errorVerbose": "insert circulating supply for tipset ({bafy2bzaceasvizcp2v3ms6hgs4e6dtuff466ushzmckubgjyxcyhk7ihi5yyk,bafy2bzacedfcucf4yraydee46v2zcexvrpwihl3jb3qcoghbcocezb5s7ttgi}):\n    github.com/filecoin-project/lotus/cmd/lotus-chainwatch/syncer.(*Syncer).storeCirculatingSupply\n        /Users/lanzafame/src/github.com/filecoin-project/lotus/cmd/lotus-chainwatch/syncer/sync.go:292\n  - pq: duplicate key value violates unique constraint \"chain_economics_pk\""}
```

File: cmd/lotus-chainwatch/syncer/sync.go (hunk 2 above)
Comment (placer14) [reply cmd/lotus-chainwatch/syncer/sync.go:286]: To make the data being added with the `db.Exec` call more obvious. (Pretend this is a prepare. :P )

**Comment by placer14** [comment]
@lanzafame Thanks for raising it. The error is harmless, though surprising that chainwatch is attempting to store a specific tipset more thank once. This behavior is temporary and will be refactor shortly, but thanks for the heads up. Can you add any context about any unusual starting/stoping of chainwatch? or is chainwatch hitting that error case through normal operation?

**Comment by lanzafame** [comment]
Normal operation on a fresh database.

**Pull request merged**